### PR TITLE
Fix `--no-emit-workspace` with `--all-packages` on single-member workspaces

### PR DESCRIPTION
**Pull request by zanieb**
Closes https://github.com/astral-sh/uv/issues/18070

Fixes an issue where `--no-emit-workspace --all-packages` would incorrectly emit `-e .` (the workspace root) when the workspace contained only a single member at the root.

When a workspace contains a single member at the root, that member is not included in the lockfile's `members` set (it's empty). The `--all-packages` flag uses `InstallTarget::Workspace`, which returned `None` for `project_name()`. This meant `include_package()` had no way to identify the root project as a workspace member — neither the `project_name` check nor the `members.contains()` check could match it, so it passed through the `--no-emit-workspace` filter.

### CODE REVIEW

File: crates/uv/tests/it/export.rs [hunk 1]
```diff
@@ -2267,6 +2265,58 @@ fn requirements_txt_no_emit() -> Result<()> {
...
+        requires-python = ">=3.12"
+        dependencies = ["anyio==3.7.0"]
+
+        [build-system]
+        requires = ["setuptools>=42"]
+        build-backend = "setuptools.build_meta"
+        "#,
+    )?;
+
+    context.lock().assert().success();
```
Comment (zanieb) [inline crates/uv/tests/it/export.rs]: There's no need to lock ahead of time, we should drop this

File: crates/uv/src/commands/project/install_target.rs [hunk 2]
```diff
@@ -95,7 +95,17 @@ impl<'lock> Installable<'lock> for InstallTarget<'lock> {
         match self {
             Self::Project { name, .. } => Some(name),
             Self::Projects { .. } => None,
-            Self::Workspace { .. } => None,
+            Self::Workspace { lock, .. } => {
+                // If the workspace contains a single member at the root, it may be omitted from
```
Comment (zanieb) [inline crates/uv/src/commands/project/install_target.rs]: This is not a "may" this is a "will", right?

**Comment by konstin** [comment]
> When a workspace contains a single member at the root, that member is not included in the lockfile's `members` set (it's empty).

Why is that, shouldn't we instead ensure that the members list is accurate?

**Comment by zanieb** [comment]
It's intentional behavior https://github.com/astral-sh/uv/blob/50eb601d80a4b34e29cd92c22f343f2dc3ed30b7/crates/uv/src/commands/project/lock_target.rs#L151-L156 and has been longstanding https://github.com/astral-sh/uv/pull/6091

I don't think we should change it, since it'll churn lock files. If we see more problems, we can consider it.

**Comment by zanieb** [comment]
I think it's intended to avoid a noisy members manifest when you're not actually using a workspace.

File: crates/uv/tests/it/export.rs [hunk 3]
```diff
@@ -2267,6 +2265,56 @@ fn requirements_txt_no_emit() -> Result<()> {
...
+        r#"
+        [project]
+        name = "project"
+        version = "0.1.0"
+        requires-python = ">=3.12"
+        dependencies = ["anyio==3.7.0"]
+
+        [build-system]
+        requires = ["setuptools>=42"]
+        build-backend = "setuptools.build_meta"
```
Comment (konstin) [inline crates/uv/tests/it/export.rs]: Here we don't build the project, We should use uv_build for all new test packages, it's faster and uses no network for the cases where we do.

**Comment by konstin** [comment]
That's a confusing edge case to do this on the `Workspace` level instead of the lockfile boundary.
